app/test_main.py

```diff
@@ -2,0 +1,30 @@
+from unittest.mock import patch
+from app.main import can_access_google_page
+
+
+def test_can_access_google_page_accessible():
+    url = "https://www.google.com"
+    with patch("app.main.valid_google_url", return_value=True), \
+         patch("app.main.has_internet_connection", return_value=True):
+        assert can_access_google_page(url) == "Accessible", "Expected 'Accessible' when both conditions are True"
+
+
+def test_can_access_google_page_invalid_url():
+    url = "https://www.invalid-url.com"
+    with patch("app.main.valid_google_url", return_value=False), \
+         patch("app.main.has_internet_connection", return_value=True):
+        assert can_access_google_page(url) == "Not accessible", "Expected 'Not accessible' when URL is invalid"
+
+
+def test_can_access_google_page_no_internet():
+    url = "https://www.google.com"
+    with patch("app.main.valid_google_url", return_value=True), \
+         patch("app.main.has_internet_connection", return_value=False):
+        assert can_access_google_page(url) == "Not accessible", "Expected 'Not accessible' when no internet connection"
+
+
+def test_can_access_google_page_not_accessible():
```

Comment:
The test function name `test_can_access_google_page_not_accessible` could be more descriptive. Consider renaming it to something like `test_invalid_url_no_connection` to better describe the scenario being tested.